libflux/src/flux/semantic/builtins.rs

```diff
@@ -301,6 +301,178 @@ pub fn builtins() -> Builtins<'static> {
+                    ) -> [t1]
+                "#),
+                "distinct" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "drop" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t1: Row, t2: Row (
+                        <-tables: [t0],
+                        ?fn: (column: string) -> bool,
+                        ?columns: t1
+                    ) -> [t2]
+                "#),
+                "duplicate" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        column: string,
+                        as: string
+                    ) -> [t1]
+                "#),
+                "duration" => Node::Builtin("forall [t0] (v: t0) -> duration"),
+                "elapsed" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?unit: duration,
+                        ?timeColumn: string,
+                        ?columnName: string
+                    ) -> [t1]
+                "#),
+                "exponentialMovingAverage" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        n: int
+                    ) -> [t1]
+                "#),
+                "false" => Node::Builtin("forall [] () -> bool"),
+                "fill" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t2: Row (
+                        <-tables: [t0],
+                        ?column: string,
+                        value: [t1],
+                        usePrevious: bool
+                    ) -> [t2]
+                "#),
+                "filter" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t2: Row (
+                        <-tables: [t0],
+                        fn: (r: t1) -> bool
+                    ) -> [t2]
+                "#),
+                "first" => Node::Builtin(r#"
+                    forall [t0] where t0: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t0]
+                "#),
+                "float" => Node::Builtin("forall [t0] (v: t0) -> float"),
+                "getColumn" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        table: [t0],
+                        column: string
+                    ) -> [t1]
+                "#),
+                "getRecord" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        table: [t0],
+                        idx: int
+                    ) -> [t1]
+                "#),
```

Comment:
`getRecord` returns a single record vs a list of records like the rest of our transformations. So its type signature should be:

    forall [t0] where t0: Row (<-table: [t0], idx: int) -> t0

Reply:
That's what I would have thought, based on the *name*, but there was something I was reading that made it seem...not that.

Reply:
@jlapacik What about `tableFind`? The current type for that one is:

```rust
                "tableFind" => Node::Builtin(r#"
                    forall [t0, t1] where t0: Row, t1: Row (
                        <-tables: [t0],
                        fn: (key: t1) -> bool
                    ) -> t0
                "#),
```

Should that read ` -> [t0]` or should we use double lists everywhere else, (i.e. `[[t0]]`)? 

Using double lists would mean `tableFind` is `(tables: [[t0]]) -> [t0]` and `getRecord` is `(table:[t0]) -> t0`. 


Alternatively we can just continue to ignore the table concept and update `tableFind` to be `(tables: [t0]) -> [t0]`

Thoughts?

Reply:
You're right @nathanielc, that signature for `tableFind` is incorrect. I would prefer to ignore the table concept for right now and just update `tableFind` to be:

    (<-tables: [t0]) -> [t0]

Reply:
@jlapacik Sounds good.

@rockstar Can you make the change to the `tableFind` function as part of this PR?

Reply:
I don't mind updating it. I'm glad to see that I'm not the only one getting a little lost in the woods typing these builtins.